src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -803,261 +825,266 @@ void func_8089BD28(EnDinofos* this, PlayState* play) {
     this->actor.bgCheckFlags &= ~BGCHECKFLAG_GROUND;
     this->actor.speed = 8.0f;
     this->actor.velocity.y = 16.0f;
     Actor_PlaySfx(&this->actor, NA_SE_EN_RIZA_JUMP);
-    this->unk_290 = 0;
-    this->unk_292 = -1;
-    this->actionFunc = func_8089C2A8;
+    this->attackTimer = 0;
+    this->timer2 = -1;
+    this->actionFunc = EnDinofos_JumpSlash;
 }
 
-void func_8089C2A8(EnDinofos* this, PlayState* play) {
+/**
+ * The Dinolfos will always swing
+ * its knife when it begins falling from jumping over the player.
+ */
+void EnDinofos_JumpSlash(EnDinofos* this, PlayState* play) {
     SkelAnime_Update(&this->skelAnime);
-    this->unk_290++;
+    this->attackTimer++;
     Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x1800);
+
     if ((this->actor.velocity.y < 0.0f) && (this->skelAnime.animation == &gDinolfosJumpAnim)) {
-        this->colliderQuad.base.atFlags |= AT_ON;
+        this->knifeCollider.base.atFlags |= AT_ON;
         this->colliderJntSph.base.acFlags |= AC_ON;
         Animation_Change(&this->skelAnime, &gDinolfosAttackAnim, 1.0f, 7.0f, 13.0f, ANIMMODE_ONCE, -2.0f);
     }
 
     if (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
-        this->colliderQuad.base.atFlags &= ~AT_ON;
-        func_8089C398(this);
+        this->knifeCollider.base.atFlags &= ~AT_ON;
+        EnDinofos_SetupLand(this);
     }
 }
 
-void func_8089C398(EnDinofos* this) {
-    if (this->actionFunc != func_8089C2A8) {
+void EnDinofos_SetupLand(EnDinofos* this) {
+    if (this->actionFunc != EnDinofos_JumpSlash) {
         Animation_PlayOnce(&this->skelAnime, &gDinolfosLandAnim);
     } else {
         this->skelAnime.endFrame = Animation_GetLastFrame(&gDinolfosAttackAnim);
     }
 
     if (this->actor.speed < 0.0f) {
-        this->unk_290 = 1;
+        this->isJumpingBackward = true;
     } else {
-        this->unk_290 = 0;
+        this->isJumpingBackward = false;
     }
 
     this->actor.speed = 0.0f;
-    func_8089AD70(this);
+    EnDinofos_EnableBumperCollision(this);
     Actor_PlaySfx(&this->actor, NA_SE_EN_BOMCHU_WALK);
-    this->actionFunc = func_8089C44C;
+    this->actionFunc = EnDinofos_Land;
 }
 
-void func_8089C44C(EnDinofos* this, PlayState* play) {
+void EnDinofos_Land(EnDinofos* this, PlayState* play) {
     if (SkelAnime_Update(&this->skelAnime)) {
-        if ((this->unk_290 == 1) && (this->actor.xzDistToPlayer < 280.0f) && func_8089A968(this) &&
-            (Rand_ZeroOne() < 0.6f)) {
-            func_8089CA14(this);
+        if ((this->isJumpingBackward == true) && (this->actor.xzDistToPlayer < 280.0f) &&
+            EnDinofos_IsFacingPlayer(this) && (Rand_ZeroOne() < 0.6f)) {
+            EnDinofos_SetupStartBreatheFire(this);
         } else {
-            func_8089A9B0(this, play);
+            EnDinofos_ChooseAction(this, play);
         }
     }
 }
 
-void func_8089C4F8(EnDinofos* this) {
+void EnDinofos_SetupSlashFromGround(EnDinofos* this) {
```

Comment:
```suggestion
void EnDinofos_SetupSlash(EnDinofos* this) {
```
We can already distinguish from the other type of slash becuase it has `Jump`, so again not sure we really need the `FromGround` here.